### PR TITLE
Add versioneer and setup.py to codecov ignore

### PR DESCRIPTION
**Pull request by jbusecke**
- [ ] Passes `isort . && black . && mypy . && flake8`
 - [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

I noticed that our code coverage is very low and it seems this is mostly a result of codecov checking both `setup.py` and `versioneer.py`

image

This PR adds both files to the ignore in `codecov.yml`. Any thoughts on this?

### CODE REVIEW

**Comment by dcherian** [comment]
Great  :+1:

**Comment by jbusecke** [comment]
If nobody is against this, ill merge this soon.

File: codecov.yml [hunk 1]
```diff
@@ -13,4 +13,8 @@ coverage:
         target: 95
         informational: true
     patch: off
-    changes: off
\ No newline at end of file
+    changes: off
+    
+ignore:
+  - "setup.py"
```
Comment (andersy005) [inline codecov.yml:19]: You may also want to exclude `tests` as well:

```suggestion
  - "setup.py"
  - "xgcm/tests/*.py"
```

**Review by rabernat — APPROVED** [review]

File: codecov.yml (hunk 1 above)
Comment (andersy005) [reply codecov.yml:19]: Never mind :) Just found out that tests are excluded in `.coveragerc`. How about consolidating codecov config files into one file by moving the contents of `.coveragerc` into `codecov.yml`?

Comment (jbusecke) [reply codecov.yml:19]: Thanks a lot for the suggestion. I have moved the ignore paths over and deleted `.coveragerc`, I did not find equivalents to the other options [in the codecov docs](https://docs.codecov.io/docs/codecovyml-reference), do you have an idea if I need those?

I mean these lines to be specific:
```

[run]
branch = True

[report]
exclude_lines =
    if self.debug:
    pragma: no cover
    raise NotImplementedError
    if __name__ == .__main__.:
ignore_errors = True
```

Comment (andersy005) [reply codecov.yml:19]: It's my understanding that any line that ends with `# pragma: no cover` is automatically excluded by codecov and as a result you don't need to include it in config file (please correct me if I am wrong or you are getting different behavior).

Regarding `ignore_errors`,  what does it do? I've never had to use it.

Comment (jbusecke) [reply codecov.yml:19]: As far as I can tell it seems to work on codecov. I would go ahead and merge.

Comment (andersy005) [reply codecov.yml:19]: Sounds good!

**Comment by jbusecke** [comment]
Thanks for the help @andersy005

**Pull request merged**